org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/IQuickFixProvider.java

```diff
@@ -33,10 +37,22 @@ public interface IQuickFixProvider {
 	 *
 	 * @param options
 	 *            Contextual action options
-	 * @param diagnostic
-	 *            the diagnostic
+	 * @param fixMethods
+	 *            a list of methods that create the resolution
 	 * @return 0..n resolutions for the given issue
 	 */
-	List<DiagnosticResolution> getResolutions(Options options, Diagnostic diagnostic);
+	List<DiagnosticResolution> getResolutions(Options options, List<Method> fixMethods);
```

Comment:
The fact that this quickFixProvider is implemented based on reflection and annotated methods, should not become part of the interface. I'm afraid we'll need to find another solution to avoid the expensive part of the document parsing.